src/activity_actor.cpp

```diff
@@ -4377,6 +4378,145 @@ std::unique_ptr<activity_actor> haircut_activity_actor::deserialize( JsonIn & )
     return haircut_activity_actor().clone();
 }
 
+rummage_pocket_activity_actor::rummage_pocket_activity_actor( const item_location &iloc,
+        const action act_kind, const tripoint &tgt )
+    : kind( act_kind ), target( tgt )
+{
+    item_loc.emplace_back( iloc, iloc->count() );
+}
+
+void rummage_pocket_activity_actor::start( player_activity &act, Character &who )
+{
+
+    int moves = 0;
+
+    for( const std::pair<item_location, int> &item : item_loc ) {
+        moves += item.first.obtain_cost( who );
+    }
+    act.moves_total = moves;
+    act.moves_left = moves;
+}
+
+void rummage_pocket_activity_actor::do_turn( player_activity &, Character &who )
+{
+    who.add_msg_if_player( _( "You rummage through your pockets" ) );
+}
+
+void rummage_pocket_activity_actor::finish( player_activity &act, Character &who )
+{
+    // some function calls in the switch block spawn activities e.g.
+    // avatar::read spawns an ACT_READ activity, so we need to set
+    // this one to null before calling them
+    act.set_to_null();
+
+    if( !item_loc.empty() ) {
+        item_location it_loc = item_loc.front().first;
+        //Adding the moves_total here so that double the cost isn't payed for doing an action
+        // there should be a better way of doing this, but I don't know it.
+        who.moves += act.moves_total;
+        switch( kind ) {
+            case action::apply_use:
+                avatar_action::use_item( *who.as_avatar(), it_loc );
+                return;
+            case action::eat:
+                avatar_action::eat( *who.as_avatar(), it_loc );
+                return;
+            case action::plthrow:
+                avatar_action::plthrow( *who.as_avatar(), it_loc );
+                return;
+            case action::read: {
+                avatar &player_character = get_avatar();
+                if( it_loc->type->can_use( "learn_spell" ) ) {
+                    item spell_book = *it_loc.get_item();
+                    spell_book.get_use( "learn_spell" )->call(
+                        player_character, spell_book, spell_book.active, player_character.pos() );
+                } else {
+                    player_character.read( it_loc );
+                }
+                return;
+            }
+            case action::wear: {
+                avatar &player_character = get_avatar();
+                player_character.wear( it_loc );
+                return;
+            }
+            case action::wield:
+                who.as_avatar()->wield( it_loc );
```

Comment:
I am not sure skipping game::wield is correct. game::wield has a number of sanity checks on what you're wielding - though perhaps they are redundant with avatar::wield.

Reply:
I guess it really isn't, after thinking about it, what if it isn't an Avatar wielding something, but a NPC or something else? The implementation should be generic. I'll revert these changes

Reply:
So it shouldn't be `who.as_avatar()->wield(it_loc)` but `who.wield(*it_loc)`, if we decide to keep using avatar:: or npc:: wield, that is.